### ISSUE TITLE
Error when running tensorflow.python.ops.gen_array_ops.pad_v2

### ISSUE THREAD

**Issue by dmc1778** [tensorflow/tensorflow#59365]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

No

### Source

binary

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
Probably due to negative or empty input
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import gen_array_ops
try:
  arg_0_tensor = tf.saturate_cast(tf.random.uniform([], minval=0, maxval=2, dtype=tf.int64), dtype=tf.uint64)
  arg_0 = tf.identity(arg_0_tensor)
  arg_1 = []
  arg_2 = -992
  out = gen_array_ops.pad_v2(arg_0,arg_1,arg_2,)
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
2023-01-21 13:38:39.787930: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-21 13:38:39.902000: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-01-21 13:38:40.466321: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 13:38:40.466519: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 13:38:40.466528: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-01-21 13:38:41.067843: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-21 13:38:41.096228: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 13:38:41.096244: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2023-01-21 13:38:41.096482: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Error:<class 'tensorflow.python.framework.ops.EagerTensor'> returned a result with an error set

```
```
</details>

**Comment by gaikwadrahul8** [comment]
Hi, @nimashiri 

Apologize for the delay and I was able to replicate the issue without any error and it's working as expected, I have added [gist file ](https://colab.research.google.com/gist/gaikwadrahul8/126f3d9da8b7bd4b1c7f0236ae32019b/-59365-pad_v2.ipynb)for your reference and in your case you're passing the zero-dimension input with zero rank and also datatype was different in your case so make sure that tensor datatypes of input and padding should be same 

Please refer our [official documentation](https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/pad-v2) for `tensorflow::ops::PadV2`

I have also replicated on `Ubuntu 22.04` with `Tensorflow==2.10` and it's working as expected with correct arguments with correct datatypes, for you reference I have added screenshot below

![image](https://user-images.githubusercontent.com/115997457/215264549-ef55acdf-7dd2-4f9b-a1f9-f73be8f0d65f.png)

If issue still persists please let us know? or Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? Thank you!

**Comment by gaikwadrahul8** [comment]
Hi, @SuryanarayanaY 

Could you please look into this issue ? Thank you!

**Comment by mihaimaruseac** [comment]
Closing as fixed. No credit
